### PR TITLE
[MM-55304] Migrate "server/channels/store/searchlayer/user_layer.go" to use GenericStoreResult

### PR DESCRIPTION
**Pull request by Paul-Stern**
#### Summary
Make use of GenericStoreResult in "server/channels/store/searchlayer/user_layer.go".

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/25309
Jira https://mattermost.atlassian.net/browse/MM-55304

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/store/searchlayer/user_layer.go [hunk 1]
```diff
@@ -135,14 +135,14 @@ func (s *SearchUserStore) autocompleteUsersInChannelByEngine(engine searchengine
 	if result.NErr != nil {
 		return nil, errors.Wrap(result.NErr, "failed to get user profiles by ids")
 	}
-	inUsers := result.Data.([]*model.User)
+	inUsers := result.Data
 	autocomplete.InChannel = inUsers
```
Comment (hanzei) [inline server/channels/store/searchlayer/user_layer.go:139]: I think we can simplify here
```suggestion
	autocomplete.InChannel = result.Data
```

File: server/channels/store/searchlayer/user_layer.go [hunk 2]
```diff
@@ -135,14 +135,14 @@ func (s *SearchUserStore) autocompleteUsersInChannelByEngine(engine searchengine
...
 
 	result = <-nuchan
 	if result.NErr != nil {
 		return nil, errors.Wrap(result.NErr, "failed to get user profiles by ids")
 	}
-	outUsers := result.Data.([]*model.User)
+	outUsers := result.Data
 	autocomplete.OutOfChannel = outUsers
```
Comment (hanzei) [inline server/channels/store/searchlayer/user_layer.go:146]: Same here
```suggestion
	autocomplete.OutOfChannel = result.Data
```

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by streamer45 — APPROVED** [review]
Nice!

**Pull request merged**